cpp/src/copying/copy.cpp

```diff
@@ -129,7 +129,7 @@ std::unique_ptr<column> allocate_like(column_view const& input,
                                   size,
                                   rmm::device_buffer(size * size_of(input.type()), stream, mr),
                                   detail::create_null_mask(size, allocate_mask, stream, mr),
```

Comment:
Is it still legal for `detail::create_null_mask` to accept `UNINITIALIZED`? I think that this `allocate_like` is permitting `UNINITIALIZED` as a value but then claims the null count is zero.

Ref: https://github.com/davidwendt/cudf/blob/f115335590990267333cf07cf6aa8c35ceaa8645/cpp/src/bitmask/null_mask.cu#L91-L95

Reply:
Yes, it is still legal to create an uninitialized mask and then the null-count would be uninitialized as well. Technically the entire output column from `allocate_like` is uninitialized (the data buffer is garbage too) so this is no different than what we have today. The caller still needs to fill in the data and the null-mask/count with real values.

Reply:
I looked at the [docs for this function](https://docs.rapids.ai/api/libcudf/nightly/group__column__copy.html#ga735c8dd32ebfbf05638095afa51eba9d). The docs are clear that the data is uninitialized. It would be a bit clearer to note the requirements for setting a null count if the null mask is uninitialized, too.